app/src/main/java/com/example/datarsd1/icebreaker/QuizActivity.java

```diff
@@ -0,0 +1,207 @@
+                askQuestion();
+            }
+        };
+
+        askQuestion();
+    }
+
+    protected void askQuestion()
+    {
+        t1.setText("Score: "+score);
+        Question q = QuestionGenerator.makeQuestion();
+
+        b1.setText(q.getOptions()[0]);
+        b2.setText(q.getOptions()[1]);
+        b3.setText(q.getOptions()[2]);
+        b4.setText(q.getOptions()[3]);
+        answer = q.getName();
+
+        if (cheat)
+        {
+            if (b1.getText().toString().equalsIgnoreCase(answer))
+                {b1.setBackgroundResource(R.drawable.cheatbuttonshape); b1.setTextColor(getResources().getColor(R.color.buttonColor));}
+            else
+                {b1.setBackgroundResource(R.drawable.buttonshape); b1.setTextColor(getResources().getColor(R.color.textColor));}
+
+            if (b2.getText().toString().equalsIgnoreCase(answer))
+                {b2.setBackgroundResource(R.drawable.cheatbuttonshape); b2.setTextColor(getResources().getColor(R.color.buttonColor));}
+            else
+                {b2.setBackgroundResource(R.drawable.buttonshape); b2.setTextColor(getResources().getColor(R.color.textColor));}
+
+            if (b3.getText().toString().equalsIgnoreCase(answer))
+                {b3.setBackgroundResource(R.drawable.cheatbuttonshape); b3.setTextColor(getResources().getColor(R.color.buttonColor));}
+            else
+                {b3.setBackgroundResource(R.drawable.buttonshape); b3.setTextColor(getResources().getColor(R.color.textColor));}
+
+            if (b4.getText().toString().equalsIgnoreCase(answer))
+                {b4.setBackgroundResource(R.drawable.cheatbuttonshape); b4.setTextColor(getResources().getColor(R.color.buttonColor));}
+            else
+                {b4.setBackgroundResource(R.drawable.buttonshape); b4.setTextColor(getResources().getColor(R.color.textColor));}
+        }
+
+        b1.setText(q.getOptions()[0]);
+        b2.setText(q.getOptions()[1]);
+        b3.setText(q.getOptions()[2]);
+        b4.setText(q.getOptions()[3]);
+
+        int profile = getResources().getIdentifier(answer.toLowerCase().replaceAll("\\s+", ""), "drawable", getPackageName());
+        ImageView pic = (ImageView) findViewById(R.id.imageView);
+        pic.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Intent contactIntent = new Intent(ContactsContract.Intents.Insert.ACTION);
+                contactIntent.setType(ContactsContract.RawContacts.CONTENT_TYPE);
+                contactIntent.putExtra(
+                        ContactsContract.Intents.Insert.NAME, answer
+                );
+                startActivity(contactIntent);
+            }
+        });
+        pic.setImageResource(profile);
+
+        timer.start();
+    }
+
+    public void onClick(View v)
+    {
+        timer.cancel();
+        score++;
```

Comment:
A little confusing for those reading the code when you increment and then decrement to not change the score!